packages/js/components/src/transient-notices/index.js

```diff
@@ -6,18 +6,17 @@ import classnames from 'classnames';
 import { OPTIONS_STORE_NAME, USER_STORE_NAME } from '@woocommerce/data';
 import PropTypes from 'prop-types';
 import { useDispatch, useSelect } from '@wordpress/data';
-import { useEffect } from '@wordpress/element';
+import { createElement, useEffect } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import SnackbarList from './snackbar/list';
-import './style.scss';
 
 const QUEUE_OPTION = 'woocommerce_admin_transient_notices_queue';
 const QUEUED_NOTICE_FILTER = 'woocommerce_admin_queued_notice_filter';
 
-function TransientNotices( props ) {
+export function TransientNotices( props ) {
```

Comment:
I have mixed feelings on bringing `TransientNotices` into the components package.  While I think this might be reusable, I think we should strive to not add anything around our data layer into generic components packages.

I know that we have 2 other components (`RichTextEditor` and `Plugins`) that do this, but IMO these should also avoid this if possible.

I think if we create props for `onRemove`, `onDequeue`, `onCreate`, and `notices` we could remove the data store and update option calls in this component in addition to making it more testable and reusable.